pkg/kubelet/kubelet.go

```diff
@@ -522,7 +507,7 @@ func NewMainKubelet(
 
 	klet.volumeManager, err = volumemanager.NewVolumeManager(
 		enableControllerAttachDetach,
-		hostname,
+		nodeName,
```

Comment:
The volume manager expects a nodeName and uses it to query the API server to verify that the attach detach controller has attached the volume to the node. When the hostname does not matches the nodeName, this lookup fails eventually failing volume mounting.